src/Raven.Server/ServerWide/Maintenance/ClusterMaintenanceSupervisor.cs

```diff
@@ -242,29 +249,39 @@ public Task StartListening(TcpConnectionInfo tcpConnection)
 
             private async Task<Stream> ConnectToClientNodeAsync(TcpConnectionInfo tcpConnectionInfo)
             {
-                var connection = await ConnectAndGetNetworkStreamAsync(tcpConnectionInfo);
-                using (_contextPool.AllocateOperationContext(out JsonOperationContext ctx))
-                using (var writer = new BlittableJsonTextWriter(ctx, connection))
+                try
                 {
-                    WriteOperationHeaderToRemote(writer);
-                    using (var responseJson = await ctx.ReadForMemoryAsync(connection, _readStatusUpdateDebugString + "/Read-Handshake-Response", _token))
+                    var connection = await ConnectAndGetNetworkStreamAsync(tcpConnectionInfo);
+                    using (_contextPool.AllocateOperationContext(out JsonOperationContext ctx))
+                    using (var writer = new BlittableJsonTextWriter(ctx, connection))
                     {
-                        var headerResponse = JsonDeserializationServer.TcpConnectionHeaderResponse(responseJson);
-                        switch (headerResponse.Status)
+                        WriteOperationHeaderToRemote(writer);
+                        using (var responseJson = await ctx.ReadForMemoryAsync(connection, _readStatusUpdateDebugString + "/Read-Handshake-Response", _token))
                         {
-                            case TcpConnectionHeaderResponse.AuthorizationStatus.Success:
-                                //All good nothing to do
-                                break;
-                            default:
-                                throw new UnauthorizedAccessException(
-                                    $"Node with ClusterTag = {ClusterTag} replied to initial handshake with authorization failure {headerResponse.Status}");
+                            var headerResponse = JsonDeserializationServer.TcpConnectionHeaderResponse(responseJson);
+                            switch (headerResponse.Status)
+                            {
+                                case TcpConnectionHeaderResponse.AuthorizationStatus.Success:
+                                    //All good nothing to do
+                                    break;
+                                default:
+                                    throw new UnauthorizedAccessException(
+                                        $"Node with ClusterTag = {ClusterTag} replied to initial handshake with authorization failure {headerResponse.Status}");
+                            }
                         }
+
+                        WriteClusterMaintenanceConnectionHeader(writer);
                     }
 
-                    WriteClusterMaintenanceConnectionHeader(writer);
+                    return connection;
                 }
+                catch (Exception)
+                {
+                    if (_cts.IsCancellationRequested) //don't care because the server is disposing
+                        return Stream.Null;
```

Comment:
You are catching _all_ errors here, not just `ObjectDisposedException` or `OperationCancelledException`. That can mask errors.
And it is better to let the calling code make decisions about this.